pkg/roachprod/roachprod.go

```diff
@@ -3026,40 +3026,48 @@ func FetchLogs(
 			if err != nil {
 				return err
 			}
-
+			// Found logs and corresponding nodes, which contain at least one log directory.
 			logDirs := make(map[string]struct{})
+			nodes := install.Nodes{}
+
 			for _, r := range results {
 				if r.Err != nil {
 					l.Printf("will not fetch logs for n%d due to error: %v", r.Node, r.Err)
+					continue
 				}
-
+				nodes = append(nodes, r.Node)
 				for _, logDir := range strings.Fields(r.Stdout) {
 					logDirs[logDir] = struct{}{}
 				}
 			}
-
+			if len(logDirs) == 0 {
+				l.Printf("no log directories found")
+				return nil
+			}
+			// For each found log directory, recursively scp its contents under the destination, prefixed by node id.
+			// E.g., n1's contents will show up under 1.unredacted, etc.
 			for logDir := range logDirs {
 				dirPath := filepath.Join(destination, logDir, "unredacted")
 				if err := os.MkdirAll(filepath.Dir(dirPath), 0755); err != nil {
 					return err
 				}
-
-				if err := c.Get(ctx, l, c.Nodes, logDir /* src */, dirPath /* dest */); err != nil {
+				// Runs a recursive scp in parallel.
+				if err := c.Get(ctx, l, nodes, logDir /* src */, dirPath /* dest */); err != nil {
 					l.Printf("failed to fetch log directory %s: %v", logDir, err)
 					if ctx.Err() != nil {
 						return errors.Wrap(err, "cluster.FetchLogs")
 					}
 				}
 			}
-
-			if err := c.Run(ctx, l, l.Stdout, l.Stderr, install.WithNodes(c.Nodes), "", fmt.Sprintf("mkdir -p logs/redacted && %s debug merge-logs --redact logs/*.log > logs/redacted/combined.log", test.DefaultCockroachPath)); err != nil {
+			// Create a single redacted log.
+			if err := c.Run(ctx, l, l.Stdout, l.Stderr, install.WithNodes(nodes), "", fmt.Sprintf("mkdir -p logs/redacted && %s debug merge-logs --redact logs/*.log > logs/redacted/combined.log", test.DefaultCockroachPath)); err != nil {
```

Comment:
`test.DefaultCockroachPath` 👀 

We probably shouldn't be importing `roachtest/test` here. Just ran into a cyclical import on an unrelated import because of this.

Reply:
Oops, good find!

Reply:
Removed and added `disallowed_imports_test`.